frontend/src/hooks/useToolbarButtons.tsx

```diff
@@ -0,0 +1,132 @@
+  const observer = useRef<ResizeObserver | undefined>();
+  const [displayTimeRoomName, setDisplayTimeRoomName] = useState<boolean>(false);
+
+  const [centerButtonLimit, setCenterButtonLimit] = useState<number>(0);
+  const [rightButtonLimit, setRightButtonLimit] = useState<number>(0);
+  const buttonWidth = 60;
+
+  useEffect(() => {
+    if (toolbarRef.current && !observer.current) {
+      const throttledSetToolbarButtons = throttle(
+        () => {
+          if (
+            !(
+              timeRoomNameRef.current &&
+              toolbarRef.current &&
+              mediaControlsRef.current &&
+              overflowAndExitRef.current &&
+              rightPanelControlsRef.current
+            )
+          ) {
+            return;
+          }
+
+          const toolbarStyle = window.getComputedStyle(toolbarRef.current);
+          const toolbarPadding =
+            parseFloat(toolbarStyle.paddingLeft) + parseFloat(toolbarStyle.paddingRight);
+          const rightPanelControlsStyle = window.getComputedStyle(rightPanelControlsRef.current);
+          const rightPanelMargin = parseFloat(rightPanelControlsStyle.marginLeft);
+          const timeRoomNameStyle = window.getComputedStyle(timeRoomNameRef.current);
+          const timeRoomNameMargin = parseFloat(timeRoomNameStyle.marginRight);
+          const necessaryComponentsWidth =
+            mediaControlsRef.current.clientWidth +
+            overflowAndExitRef.current.clientWidth +
+            toolbarPadding +
+            rightPanelMargin +
+            timeRoomNameMargin;
+
+          const spaceForExtraButtons = Math.max(
+            0,
+            toolbarRef.current.clientWidth - necessaryComponentsWidth
+          );
+          const maxButtons = Math.floor(spaceForExtraButtons / buttonWidth);
+
+          // We reserve a few buttons for the right panel
+          const maxButtonsForCenter = numberOfToolbarButtons - RIGHT_PANEL_BUTTON_COUNT;
+          // If there's more buttons able to be displayed, we only display the max for the center of the toolbar
+          const toolbarCenterLimit =
+            maxButtons > maxButtonsForCenter ? maxButtonsForCenter : maxButtons;
+
+          setDisplayTimeRoomName(maxButtons > numberOfToolbarButtons + 1);
+          setCenterButtonLimit(toolbarCenterLimit);
+          setRightButtonLimit(Math.min(numberOfToolbarButtons, maxButtons));
+        },
+        300,
+        { leading: true, trailing: false }
+      );
+
+      observer.current = new ResizeObserverPolyfill(() => {
+        throttledSetToolbarButtons();
+      });
+    }
+
+    if (!(toolbarRef.current && observer.current)) {
+      return;
+    }
+
+    // The cleanup function may not be pointing to the correct object/ref by the time it executes.
+    // We keep its reference so the cleanup function runs correctly.
+    const observedToolbar = toolbarRef.current;
+
+    observer.current.observe(toolbarRef.current);
+
+    // eslint-disable-next-line consistent-return
```

Comment:
instead of disabling the lint here we can do `return undefined;` in place where you do `return;` (I counted two places). that way you won't need to disable the lint here

Reply:
actually, I think you can disregard this comment. on a second thought it's easier to stick to `// eslint-disable-next-line consistent-return` in this case